Objects/typeobject.c

```diff
@@ -780,8 +821,8 @@ PyType_Unwatch(int watcher_id, PyObject* obj)
     return 0;
 }
 
-void
-PyType_Modified(PyTypeObject *type)
+static void
+type_modified_unlocked(PyTypeObject *type)
```

Comment:
This should call `type_modified_unlocked(subclass)` instead of `PyType_Modified(subclass)`.

Should this function also assert that the type lock is held?

(GitHub annoyingly won't let me comment on the actual line because it's too far away from a modified line)